aiocache/base.py

```diff
@@ -109,18 +110,26 @@ class BaseCache:
     NAME: str
 
     def __init__(
-        self, serializer=None, plugins=None, namespace=None, key_builder=None, timeout=5, ttl=None
+        self,
+        serializer: Optional[BaseSerializer] = None,
+        plugins: Optional[List[Any]] = None,  # aiocache.plugins depends on aiocache.base
```

Comment:
Using `Any` because `aiocache.plugins` depends on `aiocache.base`.

I tried [refactoring `API` into its own module](https://github.com/pshafer-als/aiocache/blob/simplify-cache-namespace-refactor-api/aiocache/api.py), but it causes a lot of tests to fail.  At the moment I'm presuming this is `@API.register` writing to the class variables of different objects

Reply:
> At the moment I'm presuming this is @API.register writing to the class variables of different objects

That wasn't the problem.  It's an 'order of import' problem, where `aiocache.plugins` is attempting to create hooks before `aiocache.base` registers any commands.

Another approach would be to move the base classes into a separate model, so that they can be used for type annotations.  I'll give that a try to see how 'forced' it feels.

Reply:
> Another approach would be to move the base classes into a separate module, so that they can be used for type annotations.

Yes, [that approach is cleaner](https://github.com/pshafer-als/aiocache/blob/simplify-cache-namespace-with-types/aiocache/types.py). If this becomes repeatedly necessary as type annotations are added to other parts of the codebase, then we could move other base classes into the 'types' module as needed.

Reply:
For now, just import it with `if TYPE_CHECKING:`. Further refactoring can be considered in another PR if you still feel it's appropriate.